NotesBackEnd/src/Notes.Dependency/Containers/DependencyContainer.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using Notes.Contracts.Dependency;
+using Notes.Dependency.LifeTimeManagers;
+using Unity;
+using Unity.Injection;
+
+namespace Notes.Dependency.Containers
+{
+    public class DependencyContainer : IDependencyContainerRegister, IDependencyContainerResolver
+    {
+        private readonly IUnityContainer _unityContainer;
+
+        public DependencyContainer()
+        {
+            _unityContainer = new UnityContainer();
+        }
+
+        private DependencyContainer(IUnityContainer container)
+        {
+            _unityContainer = container;
+        }
+
+        public IDependencyContainerRegister RegisterType<TTo>(Func<TTo> getObjectFunc, LifetimeTypes lifetimeType)
+        {
+            var lifetimeManager = lifetimeType.GetUnityLifetimeManager();
+
+            _unityContainer.RegisterType<TTo>(lifetimeManager, new InjectionFactory(_ => getObjectFunc()));
+            
+            return this;
+        }
+
+        public IDependencyContainerRegister RegisterType<TFrom, TTo>(LifetimeTypes lifetimeType) where TTo : TFrom
+        {
+            var lifetimeManager = lifetimeType.GetUnityLifetimeManager();
+
+            _unityContainer.RegisterType<TFrom, TTo>(lifetimeManager);
+
+            return this;
+        }
+
+        public object Resolve(Type serviceType)
+        {
+            try
+            {
+                return _unityContainer.Resolve(serviceType);
+            }
+            catch (Exception)

```

Comment:
This catch is too omnipotent (it masks too many exception/errors/problems).  Only justifiable ["catch 'em all"](http://wiki.c2.com/?PokemonExceptionHandling) is at the very top-level code that handles error page in IIS or main program of a console app with TUI… 

Here, you should catch only `ResolutionFailedException` and not even every `ResolutionFailedException` - only such that belongs to (system.web) contracts that are intentionally not fulfilled - ask around about this (me, Vašek, Matúš and Rado should all have this in fresh memory 😃)